### PR TITLE
Rewrite profile import to Go

### LINKED ISSUE #3815: Rewrite /profiles/import to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/profiles/import` is handled by Perl

## Expected / new behavior:
`/profiles/import` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `POST`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3815

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Start up TO with changes in pull request and attempt to import a profile

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes does not include an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go [hunk 1]
```diff
@@ -346,3 +346,15 @@ func GetCacheGroupNameFromID(tx *sql.Tx, id int64) (tc.CacheGroupName, bool, err
 	}
 	return tc.CacheGroupName(name), true, nil
 }
+
+// GetCacheGroupNameFromID Get Cache Group name from a given ID
+func GetParameterID(tx *sql.Tx, id int64) (tc.CacheGroupName, bool, error) {
```
[the remark this answers is not in the document]
Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go]: ahh opps I actually dont even use this - forgot to remove. Good catch

File: traffic_ops/traffic_ops_golang/profile/profile_import.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	importedProfile := tc.ProfileImportRequest{}
+
+	if err := api.Parse(r.Body, inf.Tx.Tx, &importedProfile); err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New(err.Error()), nil)
```
[the remark this answers is not in the document]
Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/profile/profile_import.go]: why do i keep doing this 🤦‍♂

File: traffic_ops/traffic_ops_golang/profile/profile_import_test.go [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+			} else {
+				rows := sqlmock.NewRows(idRow)
+				if !msr.empty {
+					rows.AddRow(msr.id)
+				}
+				mock.ExpectQuery(msr.query).WillReturnRows(rows)
+			}
+			profileID, err := importProfile(&testCase.profile, db.MustBegin().Tx)
+			mock.ExpectCommit()
+			if testCase.returnedID != profileID {
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/profile/profile_import_test.go]: type mismatch: int64 and int

File: traffic_ops/traffic_ops_golang/profile/profile_import_test.go [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+						mock.ExpectQuery(msr.query).
+							WithArgs(param.Name, param.ConfigFile, param.Value).
+							WillReturnRows(rows)
+					}
+				}
+			}
+
+			mock.ExpectExec("profile_parameter").WillReturnResult(sqlmock.NewResult(1, int64(len(testCase.parameters))))
+
+			newParams, existingParams, err := importProfileParameters(int64(1), testCase.parameters, db.MustBegin().Tx)
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/profile/profile_import_test.go]: `Cannot use int64(1) (type int64) as type int in argument to importProfileParameters`

File: traffic_ops/testing/api/v14/profiles_test.go [hunk 5]
```diff
@@ -159,6 +161,69 @@ func GetTestProfiles(t *testing.T) {
...
+	}
+	parameters := append(exportResp.Parameters, newParam)
+	// Import Profile
+	importReq := tc.ProfileImportRequest{
+		Profile:    profile,
+		Parameters: parameters,
+	}
+	importResp, _, err := TOSession.ImportProfile(&importReq)
+	if err != nil {
+		t.Fatalf("error importing Profile: %v - %v\n", profileID, err)
```
Comment (ocket8888) [inline traffic_ops/testing/api/v14/profiles_test.go]:
```gotest
    profiles_test.go:206: error importing Profile: 16 - 400 Bad Request[400] - Error requesting Traffic Ops https://localhost:6443/api/1.3/profiles/import {"alerts":[{"text":"validating: checking cdn name cdn1 existence, checking profile name TestProfileImport existence","level":"error"}]}
```
Not sure what's up with that, you definitely used CDNs in your WithObjs call.

Comment (mhoppa) [reply traffic_ops/testing/api/v14/profiles_test.go]: 😕 this worked locally for me let me rerun. Thanks for testing it.

**Pull request merged**